pype/plugins/ftrack/publish/integrate_ftrack_instances.py

```diff
@@ -68,6 +69,10 @@ def process(self, instance):
                     "name": "thumbnail"  # Default component name is "main".
                 }
                 comp['thumbnail'] = True
+                if not comp.get("published_path"):
+                    comp['published_path'] = os.path.join(comp['stagingDir'],
+                                                        comp["files"])
```

Comment:
`comp["files"]` may not be string. I suggest modification:
```
comp_files = comp["files"]
if isinstance(comp_files, (tuple, list, set)):
    filename = comp_files[0]
else:
    filename = comp_files

comp['published_path'] = os.path.join(
    comp['stagingDir'], filename 
)
```
It's not "robust" safe, but safer.

Side question: Should that be stored to representation? It's stored under `"published_path"` key, but it's not published. Don't think we have to solve that now but `# TODO` would be great.

Reply:
Absolutely. Forgot about it. Need to push version from more client-side testing I've done in the evening.  

About the `published_path` I agree too. It should be solved a bit lower in the code rather than here, but kinda need it deployed and then cleaned up. As per usual :)